### PR TITLE
Migrate runtab to viewplugin, method getSettings

### PR DESCRIPTION
**Pull request by bunsenstraat**
This is to expose methods to the plugin system. In this case getSettings and probably in the future more events and other data.

### CODE REVIEW

File: apps/remix-ide/src/app/udapp/run-tab.js [hunk 1]
```diff
@@ -44,6 +45,31 @@ export class RunTab extends LibraryPlugin {
     this.compilersArtefacts = compilersArtefacts
     this.networkModule = networkModule
     this.fileProvider = fileProvider
+    this.createVMAccount = this.pluginUDapp.createVMAccount
```
Comment (yann300) [inline apps/remix-ide/src/app/udapp/run-tab.js:48]: if PluginUdapp is not used elsewhere, I think just removing it would be better

Comment (bunsenstraat) [reply apps/remix-ide/src/app/udapp/run-tab.js:48]: Are you saying createVMAccount and the other methods are not used?

Comment (bunsenstraat) [reply apps/remix-ide/src/app/udapp/run-tab.js:48]: Are you saying createVMAccount and the others methods are not used? or just that the class is not needed.

Comment (yann300) [reply apps/remix-ide/src/app/udapp/run-tab.js:48]: no, just the `PluginUdapp` (the class is not needed)

File: apps/remix-ide/src/app/udapp/run-tab.js [hunk 2]
```diff
@@ -44,6 +44,43 @@ export class RunTab extends LibraryPlugin {
...
+      this.emit('newTransaction', tx, receipt)
+    })
+  }
+
+  getSettings () {
+    return new Promise((resolve, reject) => {
+      if (!this.container) reject(new Error('UI not ready'))
+      else {
+        resolve({
+          selectedAccount: this.container.querySelector('#txorigin').selectedOptions[0].value,
```
Comment (yann300) [inline apps/remix-ide/src/app/udapp/run-tab.js:61]: could you replace `.container.querySelector('#txorigin')` by call to `this.settingsUI`

Comment (yann300) [reply apps/remix-ide/src/app/udapp/run-tab.js:61]: same for the others settings

**Pull request closed without merging**